src/envs/playroom.py

```diff
@@ -0,0 +1,507 @@
+                door = patches.Rectangle(
+                        (x-1.0, y-5.0), 1, 10, zorder=1, linewidth=1,
+                        edgecolor='black', facecolor='brown')
+                ax.add_patch(door)
+            else:
+                door = patches.Rectangle(
+                        (x-1.0, y-5.0), 1, 1, zorder=1, linewidth=1,
+                        edgecolor='black', facecolor='brown')
+                ax.add_patch(door)
+
+        # Draw dial
+        dial_x = state.get(self._dial, "pose_x")
+        dial_y = state.get(self._dial, "pose_y")
+        dial_face = patches.Circle((dial_x, dial_y), radius=self.dial_r,
+                                   edgecolor='black', facecolor='black')
+        ax.add_patch(dial_face)
+        level = state.get(self._dial, "level")
+        dx = self.dial_r*np.sin(level*2*np.pi)
+        dy = self.dial_r*np.cos(level*2*np.pi)
+        dial_arrow = patches.Arrow(dial_x, dial_y, dx, dy, edgecolor='red',
+                                   facecolor='red')
+        ax.add_patch(dial_arrow)
+
+        # Draw table and blocks
+        table = patches.Rectangle(
+                (10, 10), 10, 10, zorder=self.table_height,
+                linewidth=1, edgecolor='black', facecolor='brown')
+        ax.add_patch(table)
+        colors = ["red", "blue", "green", "orange", "purple", "yellow",
+                  "brown", "cyan"]
+        blocks = [o for o in state if o.is_instance(self._block_type)]
+        held = "None"
+        for i, block in enumerate(sorted(blocks)):
+            x = state.get(block, "pose_x")
+            y = state.get(block, "pose_y")
+            z = state.get(block, "pose_z")
+            c = colors[i % len(colors)]  # block color
+            if state.get(block, "held") > self.held_tol:
+                assert held == "None"
+                held = f"{block.name} ({c})"
+            rect = patches.Rectangle(
+                (x - r, y - r), 2*r, 2*r, zorder=self.table_height+z,
+                linewidth=1, edgecolor='black', facecolor=c)
+            ax.add_patch(rect)
+
+        # Draw robot
+        robot_x = state.get(self._robot, "pose_x")
+        robot_y = state.get(self._robot, "pose_y")
+        fingers = state.get(self._robot, "fingers")
+        robby = patches.Circle((robot_x, robot_y), radius=1,
+                               edgecolor='black', facecolor='yellow')
+        ax.add_patch(robby)
+        rotation = state.get(self._robot, "rotation")
+        dx, dy = np.cos(rotation*np.pi/2), np.sin(rotation*np.pi/2)
+        robot_arrow = patches.Arrow(robot_x, robot_y, dx, dy,
+            edgecolor='black', facecolor='black', width=0.5)
+        ax.add_patch(robot_arrow)
+
+        plt.suptitle(f"Held: {held}, Fingers: {fingers}", fontsize=36)
+        plt.tight_layout()
+        img = utils.fig2data(fig)
+
+        return [img]
+
+    def _get_tasks(self, num_tasks: int, possible_num_blocks: List[int],
+                   rng: np.random.Generator) -> List[Task]:
```

Comment:
What is different from the superclass here?

Reply:
they're the same right now but this is to remind me to change it later